### DISCUSSION TITLE
Horn Antenna - Strange Directivity Result

### DISCUSSION THREAD

**Discussion by Elvanox** [thliebig/openEMS-Project#220] [Q&A Usage]
Hello everyone,

This is the first time that I am using openEMS, and unfortunately, I don't have enough knowledge to know all the functionnalities of OpenEMS. This is why I need you help to verify my script and analyze my simulation results. If you agree, of course

My goal is to designa horn antenna with Octave. To do that, I started with the file Horn_Antenna.m available on the tutorial. I have to make it suitable for my application, this is why I made some changes of the code.
First, the input data are a little bit different. Horn opening angle is no more an input. I replace it by the surface opening of the horn antenna (height and width). I still need to calculate automaticaly the angle of opening (described in the script). Thus, I have to change the script related to the horn opening (different points & different argument changes for AddLinPoly() function.
Secondly, I changed the CSXCAD geometry and mesh. My goal was to reduce the overall time simulation while having precise results. This is why I decided to implement 2 different grids with 2 different resolution. With that, I am able to have more resolution in the horn antenna area and less resolution next. 
Thirdly, I lowered the parameters of the FDTD simulation with NrTS at 1000 and EndCriteria at 1e-3 to have less time simulation.

Initially, my goal was to compare the results of the simulation with knowned-horn-antenna performances. I noticed that the results were very different, especially in high frequency. In my case, I am focusing on the directivity of the antenna. Some radiation pattern in high frequency simulated were very far from reality, with lot of lobes. I joined a screenshot of the directivity results for 35GHz below where the directivity for the E Plane were disappointing.
![image](https://github.com/user-attachments/assets/96fcb5d4-8dfe-4cce-a8d9-fef7075b9047)

Does some people know exactly why this kind of results can happen ? Do you know if I did something wrong in my script that influenced these bad results? 
Thanks in advance for your answers. I joined my script below :
[Script_Horn_Antenna.txt](https://github.com/user-attachments/files/16250259/Script_Horn_Antenna.txt)

**Comment by VolkerMuehlhaus** [comment]
I don't know if this is related to your pattern issue, but I would prefer an **overlap** where the side walls are touching. 
That said, your mesh looks a bit coarse to me, resulting in strong staircasing of the horn side walls.

<img width="489" alt="overlap_horn" src="https://github.com/user-attachments/assets/d9dcc628-88a1-4b8b-9a84-925370349a1c">

You also mentioned that you limited the number of time steps and increased the convergence limit to save time, but that comes at a price: your fields might not be converged, resulting in ripples in S11 and I think you might also get ripples in the pattern. So really make sure your simulation results are converged to -30dB or even better to -40dB.

**Reply by VolkerMuehlhaus** [reply]
And another finding: I now see that you lowered the mesh density to 1/5 wavelength. 

```
max_res1 = c0 / (f_stop) / unit / 10; 
max_res2 = c0 / (f_stop) / unit / 5;
```

This will fail, don't use mesh cells larger than lambda/10 !!!
For medium accuracy I usually set mesh density to 15 cells per wavelength, or lambda/20 for higher accuracy.

**Reply by Elvanox** [reply]
Thank you for your answer, really appreciate that. I made the modifications as you described. Unfortunately, I still have similar results in the patern as I showed in my previous message. Maybe due to the specific architecture of the horn antenna that I have applied. I don't know. I will try to get more familiar with the FTTD method but as a begineer in antenna design and in electromagnetic field overall, it's gonna be difficult. Thanks again for your message

**Comment by gadiLhv** [comment]
Hey there @Elvanox 

I went over your script. There were more than a few issues there, but let's focus on the ones that matter most.

1. Your mesh granularity wasn't fine enough. It's common to use 1/20-th of the wavelength.
2. Your airbox was **way** too big. Not necessarily a problem, but definitely can slow your simulation quite a bit. I assume this is why you reduced the accuracy to 1e-3, which is pretty risky when you are looking for accurate radiation patterns and S-parameters.
3. It's not necessarily you defined it wrong, but there is no need to define such a long WG port. I offered another version, but it doesn't really matter.

Now let's focus on the other issues, which aren't necessarily yours.

You placed the rear part of the antenna on the boundary. I know this has to be done, and I know why you did it. However, this creates internal reflections in your waveguide. That, you can do nothing about. Tried to make a video about it, paraview kept crashing...

**The real issue** here is that the box used to calculate the Far-field is wrong, as it passes inside the Antenna.
This could cause artifacts in your radiation pattern. 
The way to avoid this is to put an absorber inside your waveguide, and shield it. This isn't implemented in openEMS yet, however our team did already succeed in this, you can read [this discussion](https://github.com/thliebig/openEMS-Project/discussions/188).

Applying some simple fixes for the first 3 points above, I got this rad-pat.
![image](https://github.com/user-attachments/assets/de421d4c-946c-4e79-9a2e-19bf39b46014)

[test_horn.txt](https://github.com/user-attachments/files/16366489/test_horn.txt)

**Reply by gadiLhv** [reply]
Here, managed to spew out the video.


https://github.com/user-attachments/assets/bab908a7-a733-4f2f-ac61-b2aa57e0b2d3
